components/blocks/SiteFooter/LanguageSelector/LanguageSelector.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <select
+    v-model="selected"
+    class="language-selector"
+    role="listbox"
+    @change="onChange"
+  >
+    <option v-for="locale in locales" :key="locale.code" :value="locale.code">
+      {{ locale.name }}
+    </option>
+  </select>
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  ref,
+  useContext,
+} from '@nuxtjs/composition-api';
+import { Locale } from '@/types';
+
+export default defineComponent({
+  name: 'LanguageSelector',
+  setup() {
+    const { i18n } = useContext();
+
+    const locales = computed(() => {
+      return i18n.locales as Locale[];
+    });
+
+    const currentLocale = computed(() => {
+      return locales.value.find(
+        (locale: Locale) => locale.code === i18n.locale,
+      );
+    });
+
+    const selected = ref(currentLocale.value?.code);
+
+    const onChange = (event: any) => {
+      i18n.setLocale(event.target.value);
+      selected.value = event.target.value;
```

Comment:
You don't need to set the selected value manually because you're already using `v-model` to do that automatically.